athena-gcs/README.md

```diff
@@ -0,0 +1,121 @@
+* **SubnetIds:** One or more Subnet IDs corresponding to the Subnet that the Lambda function can use to access your data source. (e.g. subnet1, subnet2)
+* **SecretNameOrPrefix:** The name or prefix of a set of names within Secrets Manager that this function should have access to. (e.g. gcs-*)
+* **VpcId:** The VPC ID to be attached to the Lambda function
+* **SecretNameOrPrefix** The name or prefix of a set of names within Secrets Manager that this function should have access to. (e.g. gcs-*). Please note that if you set up a required secret value without the given prefix, Athena will raise error.
+* **GcsSecretName** Name of the secret in the AWS Secret Manager
+* **GcsCredentialKey** Name of the key that contains GCS credential JSON (generated via service account key). Please refer to [this document](https://cloud.google.com/iam/docs/creating-managing-service-account-keys) for more details
+* **FileExtension** File extension (without period character) of the file in GCS buckets. Only one extension can be set; eitHer PARQUET or CSV (case-insensitive). These files will be parsed and shown as a Table under a database. So as you know, buckets are treated as databases. Please also note that invalid ANSI SQL characters in the file name will be replaced by an underscore character ('_'), and the file extension will be removed in the Table name
+
+### Terms
+
+* **Database Instance:** Any bucket in GCS.
+* **Handler:** A Lambda handler accessing your database instance(s). It could be metadata or a record handler.
+* **Metadata Handler:** A Lambda handler retrieves metadata from your database instance(s).
+* **Record Handler:** A Lambda handler that retrieves data records from your database instance(s).
+* **Composite Handler:** A Lambda handler retrieves metadata and data records from your database instance(s). This is recommended to be set as a Lambda function handler.
+
+### Database-specific handler parameters
+
+Database-specific metadata (extracted from GCS buckets and files inside those buckets) be used to list buckets (hence database) and tables (hence tables) and record handlers is used to fetch records and spill the data to an S3 bucket.
+
+|Handler|Class|
+|---|---|
+|Composite Handler|GCSCompositeHandler|
+|Metadata Handler|GCSMetadataHandler|
+|Record Handler|GCSRecordHandler|
+
+### Spill parameters:
+
+Lambda SDK may spill data to S3. All database instances are accessed using a single Lambda spill to the exact location.
+
+```
+spill_bucket  Spill bucket name. Required.
+spill_prefix  Spill bucket key prefix. Required.
+```
+
+### GCS Data types supported
+
+| PARQUET       | Arrow     |
+|-----------|-----------|
+| BINARY | VARCHAR|
+| BOOLEAN |BOOL|
+| DOUBLE| DOUBLE|
+| ENUM| VARCHAR|
+| FIXED_LEN_BYTE_ARRAY| DECIMAL|
+| FLOAT | FLOAT (32-bit)|
+| INT32|INT32  or DATEDAY (When Parquet’s column’s logical type is DATE)|
+| INT64 | INT64 OR Timestamp|
+| INT96 | Timestamp|
+| MAP | VARCHAR|

```

Comment:
Can we update this section. 
From GCS FUNCTIONAL TEST CASES documentation provided, it seems like we actually support struct with defined field type. 

Also, it seems like we support list as well from the test document

Reply:
read me file will be shared separately by email.